### PR TITLE
docs: update styling support for streaming

### PR DESCRIPTION
**Pull request by huozhi**
## Documentation / Examples

- [x] Make sure the linting passes by running `yarn lint`

### CODE REVIEW

File: docs/advanced-features/react-18/streaming.md [hunk 1]
```diff
@@ -68,4 +68,6 @@ Currently, data fetching within `Suspense` boundaries on the server side is not
 
 #### Styling
 
-The Next.js team is working on support for `styled-jsx` and CSS modules in streaming SSR. Stay tuned for updates.
+So far only `styled-jsx` is supported with streaming. The Next.js team is working on support for CSS modules and other CSS solutions in streaming SSR. Stay tuned for updates.
```
Comment (styfle) [inline docs/advanced-features/react-18/streaming.md:71]: Can we link to a RFC or Discussion for the updates?

File: docs/advanced-features/react-18/streaming.md [hunk 2]
```diff
@@ -68,4 +68,6 @@ Currently, data fetching within `Suspense` boundaries on the server side is not
...
+
+Notice that for React Server Components, `styled-jsx` should be only placed in client components.
```
Comment (styfle) [inline docs/advanced-features/react-18/streaming.md:73]:
```suggestion
> Note: `styled-jsx` should be only placed in client components, not server components, when using React Server Components
```

File: docs/advanced-features/react-18/streaming.md (hunk 1 above)
Comment (shuding) [inline docs/advanced-features/react-18/streaming.md:71]: We should also mention inlined styles and global solutions (global CSS import, Tailwind CSS) here.

Comment (devknoll) [inline docs/advanced-features/react-18/streaming.md:71]: `unstable_useFlushEffects` should be sufficient for any library that follows the [recommended upgrade guide](https://github.com/reactwg/react-18/discussions/110). It's just that `styled-jsx` support is built-in.

**Comment by huozhi** [comment]
@styfle @devknoll Now the built-in styling solutions (global css, css modules and styled-jsx) are supported, for other CSS solutions I think we need some time to investigate to see if they can leverage `useFlushEffects` to support them. I'll follow up that in another thread.

**Review by styfle — DISMISSED** [review]

File: docs/advanced-features/react-18/streaming.md [hunk 3]
```diff
@@ -68,4 +68,6 @@ Currently, data fetching within `Suspense` boundaries on the server side is not
 
 #### Styling
 
-The Next.js team is working on support for `styled-jsx` and CSS modules in streaming SSR. Stay tuned for updates.
+React inline styles, Global CSS, CSS modules and Next.js built-in `styled-jsx` are supported with streaming. The Next.js team is working on the guide of integrating other CSS-in-JS solutions in streaming SSR. Stay tuned for updates.
```
Comment (shuding) [inline docs/advanced-features/react-18/streaming.md:71]:
```suggestion
Inline styles, Global CSS, CSS modules and Next.js built-in `styled-jsx` are supported with streaming. The Next.js team is working on the guide of integrating other CSS-in-JS solutions in streaming SSR. Stay tuned for updates.
```

**Review by styfle — APPROVED** [review]

**Review by shuding — APPROVED** [review]

**Pull request merged**